### ISSUE TITLE
Gemfile.lock is the wrong file extension for the file's purpose. 

### ISSUE THREAD

**Issue by duncan** [bundler/bundler#694]
For better or worse, Gemfile.lock is a poor name choice. The .lock extension is a long used UNIX indication that a resource is locked for changing or update by a process. It's as entrenched a convention as .profile or the like. When I was a sysadmin, back in the stone ages, it wasn't unusual to see scripts that would remove any hinky .lock files on system startup anywhere they existed on the system. Not that those were very smart scripts, but... the expectation is that a .lock file is a transitory thing that can be vaporized.

In other words, Bundler's use of this extension is running headlong into a multiple-decade long UNIX convention. When I first saw it, it was like watching one culture clash right into another.

Not wanting to poke a hole in a naming convention without a counter proposal, here's a suggestion: use Gemfile.manifest as a formal name for this file. After all, that's what Bunder is doing when it creates it--itemizing everything that needs to go into the box and writing it down.

(btw, Bundler is good sauce. I like it.) 

-Duncan

**Comment by trans** [comment]
@indirect A reminder that non-Rails project do not check-in the lock file. As such a developer I often wish the lock file could be a hidden dot file and out of the way. But I understand that is less than ideal for Rails developers and/or if alternate Gemfiles are needed. Which is also why `Gemlock` might not be a good choice. I have seen projects that have multiple Gemfiles -- I have never done it, but I suppose it is to use/test various dependency configurations. Perhaps Bundler has improved such that is not ever necessary anymore, but if not, it needs to be considered, i.e. the extension keeps the files separate but related, e.g. `AltGemfile` and `AltGemfile.lock`.

Personally I don't see the problem with `.lock`. It is similar enough to a temporary file in that it can easily be recreated and it gets updated regularly. But if it is going to change, then perhaps `.bundle` make's the most sense.

Another idea, let a configuration option in the Gemfile itself specify the name of its lock file. For example, in a generic Ruby project I could imagine adding:

    lockfile ".bundle"

Not 100% sure if that's possible (or even desirable) but it would offer the most flexibility.

**Comment by indirect** [comment]
We've been expanding in the direction of allowing the lockfile name to be configured, and it may even be possible already. We've also shipped support for files named `gems.rb` and `gems.locked`, if present.

Ultimately, I expect that in Bundler 2.0 the Gemfile will default to `gems.rb`, and the lockfile name will be able to be configured per project.

**Comment by franklinyu** [comment]
Sorry if this looks dumb, but is there any difference in syntax between `Gemfile` and `gems.rb`, or it's merely a rename? What about `Gemfile.lock` and `gems.locked`? I didn't find `gems.rb` mentioned in the documentation at all.

**Comment by segiddins** [comment]
Same file, different name

**Comment by davecb** [comment]
My expectation is that it's a temporary file and will be released when the changes to the file without lock have been completed.  
I still write "mv x x.lock; sed 'some expression' <x.lock >x; rm lock", knowing that in Unix, _mv_ is one of the atomicity primities. 
A non-misleading name would be desirable.

**Comment by segiddins** [comment]
`gems.rb` will be the default in the next version of Bundler

**Comment by indirect** [comment]
(and once the Gemfile is instead named `gems.rb`, the locked gems list will be stored in `gems.locked`, so I think this ticket is now resolved.)

**Comment by godfat** [comment]
I might be late, but why `.rb` if it's not a Ruby file? Can we pick something else which doesn't end with `.rb` if they're not Ruby file?

**Comment by indirect** [comment]
@godfat it is literally a ruby file, though, and we "parse" it using `eval`. do you feel like it is not ruby somehow?

**Comment by godfat** [comment]
@indirect Oh, you're right. Sorry I was confused with the lock file. Not sure what I was thinking.
